performance_manager/lib/rt_trip_updates.py

```diff
@@ -91,43 +93,50 @@ def explode_stop_time_update(
 # pylint: enable=too-many-arguments
 
 
-def unwrap_tu_dataframe(events: pandas.DataFrame) -> pandas.DataFrame:
+def get_and_unwrap_tu_dataframe(
+    paths: Union[str, List[str]]
+) -> pandas.DataFrame:
     """
     unwrap and explode trip updates records from parquet files
     parquet files contain stop_time_update field that is saved as list of dicts
     stop_time_update must have fields extracted and flattened to create
     predicted trip update stop events
     """
-    # store start_date as int64
-    events["start_date"] = pandas.to_numeric(events["start_date"]).astype(
-        "int64"
-    )
-
-    # store direction_id as int64
-    events["direction_id"] = pandas.to_numeric(events["direction_id"]).astype(
-        "int64"
-    )
-
-    # store start_time as seconds from start of day int64
-    events["start_time"] = (
-        events["start_time"].apply(start_time_to_seconds).astype("int64")
-    )
-
-    # expand and filter stop_time_update column using numpy vectorize
-    # numpy vectorize offers significantly better performance over pandas apply
-    # this will return a ndarray with values being list of dicts
-    vector_explode = numpy.vectorize(explode_stop_time_update)
-    events = pandas.Series(
-        vector_explode(
-            events.stop_time_update,
-            events.timestamp,
-            events.direction_id,
-            events.route_id,
-            events.start_date,
-            events.start_time,
-            events.vehicle_id,
+    events = pandas.Series(dtype="object")
+    for batch_events in get_tu_dataframe_chunks(paths):
+        # store start_date as int64
```

Comment:
Add loop for processing trip update dataframe batches. 

100_000 default record limit for dataframe batching results in around 5-6GB of memory usage per batch

Reply:
might be a good comment on this for loop.

Reply:
will incorporate